### PR TITLE
Serialize enriched metadata - Part 1

### PR DESCRIPTION
**Pull request by thomasabraham**
## Description
Replaced GenericRecordMetadata with NakadiAvroMetadata.
The NakadiMetadata interface is converted to a DTO as well.

> Zalando ticket : team-aruha/issues/1106

## Review
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: core-services/src/main/java/org/zalando/nakadi/service/publishing/NakadiRecordMapper.java [hunk 1]
```diff
@@ -44,35 +41,26 @@ public List<NakadiRecord> fromBytesBatch(final byte[] batch) throws IOException
...
                                               final GenericRecord event) throws IOException {
         final byte[] data = EnvelopeHolder.produceBytes(
-                metadataVersion,
-                (outputStream -> {
-                    final GenericDatumWriter eventWriter = new GenericDatumWriter(metadata.getSchema());
-                    eventWriter.write(metadata, EncoderFactory.get()
-                            .directBinaryEncoder(outputStream, null));
-                }),
+                metadata.getMetadataVersion(),
+                metadata,
```
Comment (a1exsh) [inline core-services/src/main/java/org/zalando/nakadi/service/publishing/NakadiRecordMapper.java:56]: Oh.  Now I'm not sure I like the implicit way that `metadata` implements the writer interface...

File: core-services/src/main/java/org/zalando/nakadi/service/publishing/NakadiKpiPublisher.java [hunk 2]
```diff
@@ -115,29 +115,23 @@ public void publish(final Supplier<KPIEvent> kpiEventSupplier) {
         }
     }
 
-    private GenericRecord buildMetaDataGenericRecord(
-            final String eventType, final Schema metadataSchema,
+    private NakadiAvroMetadata buildMetaDataGenericRecord(
```
[the remark this answers is not in the document]
Comment (thomasabraham) [reply core-services/src/main/java/org/zalando/nakadi/service/publishing/NakadiKpiPublisher.java]: Done

**Comment by a1exsh** [comment]
:+1:

**Comment by thomasabraham** [comment]
:+1:

**Pull request merged**